File: mvp/src/main/java/com/lipeng/mvp/MvpPresenter.java
package com.lipeng.mvp;

import android.support.annotation.UiThread;

/**
 * Presenter
 * @author lipeng
 * @date 2017/12/26
 */

public interface MvpPresenter<V extends MvpView> {
    /**
     * 数据集或者事件到达此presenter时调用的方法
     * @param view 目标
     * */
    @UiThread
    void attachView(V view);

    /**
     * view销毁则会调用，通常会被<code>Activity.detachView()</code>或者<code>Fragment.onDestroy()</code>
     * 这两个方法调用
     * */
    @UiThread
    void detachView();
}

File: app/src/main/java/com/lipeng/mygithub/base/mvp/MvpPresenter.kt
package com.lipeng.mygithub.base.mvp

import android.support.annotation.UiThread

/**
 * @author big insect
 */
interface MvpPresenter <V: MvpView>{

    @UiThread
    fun attachView(v :V?)

    @UiThread
    fun detachView(isDetach :Boolean)
}